lib/concurrent/atomic_reference/direct_update.rb

```diff
@@ -27,9 +27,28 @@ def update
     # @!macro [attach] atomic_reference_method_try_update
     #
     # Pass the current value to the given block, replacing it
+    # with the block's result. Return nil if the update fails.
+    #
+    # @yield [Object] Calculate a new value for the atomic reference using
+    #   given (old) value
+    # @yieldparam [Object] old_value the starting value of the atomic reference
+    #
+    # @return [Object] the new value, or nil if update failed
+    def try_update
+      old_value = get
+      new_value = yield old_value
+
+      return unless compare_and_set old_value, new_value
+
+      new_value
+    end
```

Comment:
:+1: We could add a note to the documentation though to warn the users. If they need to distinguish `nil`, `compare_and_set` can be always used. From my side we've considered `[success, new_value]` and it's a bad idea.